zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/QuorumPeerConfig.java

```diff
@@ -399,6 +399,8 @@ public void parseProperties(Properties zkProp) throws IOException, ConfigExcepti
                 X509AuthenticationConfig.getInstance().setZnodeGroupAclOpenReadAccessPathPrefixStr(value);
             } else if (key.equals(X509AuthenticationConfig.STORE_AUTHED_CLIENT_ID)) {
                 X509AuthenticationConfig.getInstance().setStoreAuthedClientIdEnabled(value);
+            } else if (key.equals(X509AuthenticationConfig.ALLOWED_CLIENT_ID_AS_ACL_DOMAINS)) {
+                X509AuthenticationConfig.getInstance().setAllowedClientIdAsAclDomainsStr(value);
```

Comment:
This will actually not work (never worked for any of the other config keys which use `ZNODE_GROUP_ACL_CONFIG_PREFIX + something` because `NODE_GROUP_ACL_CONFIG_PREFIX = "zookeeper.X509ZNodeGroupAclProvider."` and we never use the `zookeeper.` prefix in zoo.cfg).

So, it always falls through and gets set directly as a system property in the final else block.

Reply:
Will be addressed by @PapaCharlie in a different PR.